KarhooUISDK/Screens/QuoteList/QuoteListViewController.swift

```diff
@@ -0,0 +1,272 @@
+            navigationController?.navigationBar.backItem?.title = ""
+            let appearance = UINavigationBarAppearance()
+            appearance.configureWithOpaqueBackground()
+            appearance.backgroundColor = navigationBarColor
+            appearance.setBackIndicatorImage(backArrow, transitionMaskImage: backArrow)
+            navigationController?.navigationBar.standardAppearance = appearance
+            navigationController?.navigationBar.scrollEdgeAppearance = navigationController?.navigationBar.standardAppearance
+        }
+    }
+    
+    // MARK: - State handling
+
+    private func handleStateUpdate(_ state: QuoteListState) {
+        setNavigationBarTitle(forState: state)
+        switch state {
+        case .loading:
+            self.handleLoadingState()
+        case .fetching(quotes: let quotes):
+            self.handleFetchingState(quotes: quotes)
+        case .fetched(quotes: let quotes):
+            self.handleFetchedState(quotes: quotes)
+        case .empty(reason: let reason):
+            self.handleEmptyState(reason: reason)
+        }
+    }
+    
+    private func handleLoadingState() {
+        setHeaderDisabled { [weak self] in
+            self?.tableViewCoordinator.updateQuoteListState(.loading)
+        }
+    }
+
+    private func handleFetchingState(quotes: [Quote]) {
+        setHeaderEnabled { [weak self] in
+            self?.tableViewCoordinator.updateQuoteListState(.fetching(quotes: quotes))
+        }
+    }
+
+    private func handleFetchedState(quotes: [Quote]) {
+        setHeaderEnabled { [weak self] in
+            self?.tableViewCoordinator.updateQuoteListState(.fetched(quotes: quotes))
+        }
+    }
+    
+    private func handleEmptyState(reason: QuoteListState.Error) {
+        setHeaderDisabled { [weak self] in
+            self?.tableViewCoordinator.updateQuoteListState(.empty(reason: reason))
+        }
+    }
+
+    private func setHeaderEnabled(completion: @escaping () -> Void = { }) {
+        UIView.animate(
+            withDuration: UIConstants.Duration.short,
+            animations: { [weak self] in
+                self?.quoteCategoryBarView.isHidden = false
+                self?.quoteSortView.setEnabled()
+                self?.legalDisclaimerLabel.isHidden = false
+            }
+        ) { _ in
+            completion()
+        }
+    }
+
+    private func setHeaderDisabled(completion: @escaping () -> Void = { }) {
+        UIView.animate(
+            withDuration: UIConstants.Duration.short,
+            animations: { [weak self] in
+                self?.quoteCategoryBarView.isHidden = true
+                self?.quoteSortView.setDisabled()
+                self?.legalDisclaimerLabel.isHidden = true
+            }
+        ) { _ in
+            completion()
+        }
+    }
+
+    // TODO: Prepare and update values for all possible cases
+    private func setNavigationBarTitle(forState state: QuoteListState){
+        switch state {
```

Comment:
Need to determinate and update values, especially for loading and empty state.

Reply:
For now, I will remove title for those states. We will discuss with Oliver what should be shown once he's back.